ui/pages/Project/selectors.js

```diff
@@ -82,28 +95,34 @@ export const getProjectAnalysisGroupIndividualsByGuid = createSelector(
     }), {}),
 )
 
-export const getProjectAnalysisGroupSamplesByGuid = createSelector(
-  getSamplesByGuid,
-  getProjectAnalysisGroupIndividualsByGuid,
-  (samplesByGuid, individualsByGuid) =>
-    Object.values(individualsByGuid).reduce((acc, individual) => ({
-      ...acc,
-      ...individual.sampleGuids.reduce((sampleAcc, sampleGuid) => (
-        { ...sampleAcc, [sampleGuid]: samplesByGuid[sampleGuid] }
-      ), {}),
-    }), {}),
+export const getProjectAnalysisGroupSamplesByTypes = createSelector(
```

Comment:
before, we were formatting data in a selector and then post-processing it in the render method of the component. This just makes the selector process the data so it is exactly the way the component needs it and no computation is redone if a rerender occurs. It also makes use of the already-computed `samplesByFamily`, instead of recomputing that here